### PR TITLE
Publish to Nuget.org

### PR DESCRIPTION
**Pull request by lukeryannetnz**
#31

## Added .nuspec for primary library. 
- Included references, suspect there is a better way to do this that wouldn't require updating in two places (packages.config and here). Any ideas @ddhi004?
    * Update: found that nuget can do this for you automatically based on references with the includereferencedprojects flag, which is set by appveyor with the build setting include_nuget_references: true 
    * Updated update: Found that this didn't set the version of the nuget package correctly. Followed instructions on this SO to set explicitly: http://stackoverflow.com/questions/34273570/how-to-publish-beta-nuget-packages-out-of-appveyor
* Updated appveyor build to:
    * stop building for non-master or PR branches
    * package nuspec after building
    * publish to nuget.org
    * Print the version number on the assemblies via VersionInfo.cs file.

### Possible improvements
* Could include some .config file transformations in the nuget package to set the config file up how we need it, although this is really just how the AWS library requires it.

### CODE REVIEW

**Comment by lukeryannetnz** [comment]
Check out the artifacts tab of the PR build for the nupkg :-)
https://ci.appveyor.com/project/lukeryannetnz/quartznet-dynamodb/build/artifacts

**Comment by lukeryannetnz** [comment]
screen shot 2017-02-19 at 5 55 05 pm
Versions stamped correctly on assembly within nupkg.

File: src/QuartzNET-DynamoDB/QuartzNET-DynamoDB.csproj [hunk 1]
```diff
@@ -32,6 +32,8 @@
     <WarningLevel>4</WarningLevel>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <RunCodeAnalysis>true</RunCodeAnalysis>
+    <DocumentationFile>
+    </DocumentationFile>
```
Comment (dhruv11) [inline src/QuartzNET-DynamoDB/QuartzNET-DynamoDB.csproj:36]: VS being weird?

File: appveyor.yml [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+
+assembly_info:
+  patch: true
+  file: '**\VersionInfo.*'
+  assembly_version: '$(buildVersion)'
+  assembly_file_version: '$(buildVersion)'
+  assembly_informational_version: '$(nugetVersion)'
+
+pull_requests:
+  do_not_increment_build_number: true
```
Comment (dhruv11) [inline appveyor.yml:24]: 👍

**Comment by dhruv11** [comment]
:1st_place_medal:

**Comment by dhruv11** [comment]
don't think the package made it to nuget.org

https://ci.appveyor.com/project/lukeryannetnz/quartznet-dynamodb/build/91#L994

File: src/QuartzNET-DynamoDB/QuartzNET-DynamoDB.csproj (hunk 1 above)
Comment (lukeryannetnz) [reply src/QuartzNET-DynamoDB/QuartzNET-DynamoDB.csproj:36]: Yeah :(

**Pull request merged**